### PR TITLE
Feature/companion

### PR DESCRIPTION
**Pull request by atharvagupta2003**
-Implemented companion attack
-Implemented companion defence shield
-Improved UI to match the game theme
-modified laboratory
-added test classes
Bug fix
there is still an error when you spawn some of the potions in the laboratory. A minor bug, only issue is loading in the sprite correctly, but we will fix it soon.

### CODE REVIEW

File: source/core/assets/levels/earth/main-area/entities/laboratory.json
Comment (Foref) [inline source/core/assets/levels/earth/main-area/entities/laboratory.json:1]: probably need to stringify the class path?

File: source/saves/savefile.json
Comment (Foref) [inline source/saves/savefile.json:1]: maybe delete the savefile?

Comment (atharvagupta2003) [reply source/saves/savefile.json:1]: done

**Review by aman-86 — APPROVED** [review]

**Review by zombiechook — APPROVED** [review]

**Review by DRG31 — APPROVED** [review]
Approved. Looks nice

**Pull request merged**